sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/refresh/CalculatedBackoffTime.java

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.cloud.config.refresh;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.concurrent.TimeUnit;
+
+import com.azure.spring.cloud.config.properties.AppConfigurationProviderProperties;
+
+/**
+ * Calculates the amount of time to the next refresh, if a refresh fails.
+ */
+public final class CalculatedBackoffTime {
+
+    static HashMap<String, Integer> attempts = new HashMap<>();
```

Comment:
Instead of maintaining the state of attempts in this class, is there a way to track attempts with the watch keys? This class should only track max allowed attempts.

Reply:
I am tracking here because each store has its own refresh-interval, not watch key. So, all watch keys are checked at the same time. They key here is an endpoint/"client" to track the refresh checks.

Reply:
Can the attempts be tracked with `State`? State has the watch keys for each endpoint, their refresh time, etc. It seems appropriate to store attempts there too.

Reply:
Done